tests/architecture/test_issue_300_invariants.py

```
@@ -0,0 +1,122 @@
+)
+
+
+def _spec_of(paginator: SyncPaginator[Any] | AsyncPaginator[Any]) -> PaginatedSpec[Any]:
+    """Return the ``PaginatedSpec`` a returned paginator was built from.
+
+    The engine runner returns a paginator wrapping
+    ``functools.partial(self._fetch_page, spec, args, page_size)`` (see
+    ``_runner.py`` / ``_async_runner.py``), so the spec that actually governs
+    a call is recoverable straight from the public method's return value.
+    """
+    fetch_page = paginator._fetch_page  # noqa: WPS437
```

Ruff (0.15.21)

[warning] Invalid rule code in `# noqa`: WPS437

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)